File: Color_Shooter/src/core/src/com/salvai/centrum/actors/effects/Explosion.java
package com.salvai.centrum.actors.effects;

import com.badlogic.gdx.audio.Sound;
import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.g2d.ParticleEffect;
import com.badlogic.gdx.math.Vector2;
import com.salvai.centrum.utils.Constants;

public class Explosion {
    public ParticleEffect particleEffect;
    public Vector2 position;
    public Sound sound;

    /**
     * for center ball
     *
     * @param color
     */
    public Explosion(Color color, Sound sound, boolean soundOn, ParticleEffect particleEffect) {

        this.sound = sound;
        this.particleEffect = new ParticleEffect(particleEffect);
        position = new Vector2(Constants.WIDTH_CENTER, Constants.HEIGHT_CENTER);
        this.particleEffect.getEmitters().first().setPosition(position.x, position.y);
        float[] colorF = {color.r, color.g, color.b};
        this.particleEffect.getEmitters().first().getTint().setColors(colorF);
        this.particleEffect.scaleEffect(2);
        this.particleEffect.start();
        this.particleEffect.getEmitters().shuffle();
        if (soundOn)
            this.sound.play();
    }

    /**
     * for enemies
     * @param position
     * @param color
     * @param sound
     * @param soundOn
     * @param particleEffect
     */
    public Explosion(Vector2 position, Color color, Sound sound, boolean soundOn,ParticleEffect particleEffect) {
        this.sound = sound;
        this.position = position;
        this.particleEffect = new ParticleEffect(particleEffect);
        this.particleEffect .getEmitters().first().setPosition(position.x, position.y);
        float[] colorF = {color.r, color.g, color.b};
        this.particleEffect.getEmitters().first().getTint().setColors(colorF);
        this.particleEffect.getEmitters().shuffle();
        this.particleEffect.start();
        if (soundOn)
            this.sound.play();

    }

}

File: Color_Shooter/core/src/com/salvai/centrum/actors/effects/Explosion.kt
package com.salvai.centrum.actors.effects

import com.badlogic.gdx.audio.Sound
import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.graphics.g2d.ParticleEffect
import com.badlogic.gdx.math.Vector2
import com.salvai.centrum.utils.Constants

class Explosion {
    var particleEffect: ParticleEffect
    var position: Vector2
    var sound: Sound

    /**
     * for center ball
     */
    constructor(color: Color, sound: Sound, soundOn: Boolean, particleEffect: ParticleEffect) {
        this.sound = sound
        this.particleEffect = ParticleEffect(particleEffect)
        position = Vector2(Constants.WIDTH_CENTER.toFloat(), Constants.HEIGHT_CENTER.toFloat())
        this.particleEffect.emitters.first().setPosition(position.x, position.y)
        val colorF = floatArrayOf(color.r, color.g, color.b)
        this.particleEffect.emitters.first().tint.colors = colorF
        this.particleEffect.scaleEffect(2f)
        this.particleEffect.start()
        this.particleEffect.emitters.shuffle()
        if (soundOn) this.sound.play()
    }

    /**
     * for enemies
     */
    constructor(position: Vector2, color: Color, sound: Sound, soundOn: Boolean, particleEffect: ParticleEffect) {
        this.sound = sound
        this.position = position
        this.particleEffect = ParticleEffect(particleEffect)
        this.particleEffect.emitters.first().setPosition(position.x, position.y)
        val colorF = floatArrayOf(color.r, color.g, color.b)
        this.particleEffect.emitters.first().tint.colors = colorF
        this.particleEffect.emitters.shuffle()
        this.particleEffect.start()
        if (soundOn) this.sound.play()
    }
}